# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.cpp
# Commit: 1a3bc285395f
# Full Hash: 1a3bc285395ffcc5d9281dd45d4f6c2bb918bf43
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-03-04 04:58:00
# Regressor Bug: 1861078
# File Overlap Count: 1
# Description:
#   Bug 1861078 - wasm: make metadata offsets relative to CodeBlock->codeBase. r=rhunt
#   
#   Metadata used offsets to the beginning of code segment. Change that to use
#   offsets relative to the beginning of CodeBlock. This allows us to avoid offseting
#   metadata later on and prepare the work for simplification of stackmaps.
# ==============================================================================

diff -r 66d70b730ea6 -r 1a3bc285395f js/src/wasm/WasmFrameIter.cpp
--- a/js/src/wasm/WasmFrameIter.cpp	Mon Mar 03 16:15:17 2025 +0000
+++ b/js/src/wasm/WasmFrameIter.cpp	Mon Mar 03 16:37:47 2025 +0000
@@ -1341,7 +1341,7 @@
   const CodeBlock* codeBlock = LookupCodeBlock(pc, &codeRange);
   if (codeBlock) {
     code = codeBlock->code;
-    codeBase = codeBlock->segment->base();
+    codeBase = codeBlock->base();
     MOZ_ASSERT(codeRange);
   } else if (!LookupBuiltinThunk(pc, &codeRange, &codeBase)) {
     return false;